bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -757,9 +768,12 @@ void DWARFRewriter::updateDebugInfo() {
               : std::optional<std::string>(opts::DwarfOutputPath.c_str());
       std::string DWOName = DIEBlder.updateDWONameCompDir(
           *StrOffstsWriter, *StrWriter, *CU, DwarfOutputPath, std::nullopt);
-      processSplitCU(*CU, **SplitCU, DIEBlder, *TempRangesSectionWriter,
-                     AddressWriter, DWOName, DwarfOutputPath);
+      ThreadPool.async([&, DwarfOutputPath, DWOName] {
```

Comment:
Is it necessary to specify DwarfOutputPath, DWOName? Why can't we just do [&]

Reply:
Capturing DwarfOutputPath and DWOName by reference in the async call resulted in errors where they were invalid values, likely because the ThreadPool executed it asynchronously so they were out of scope.

Reply:
Ah OK, for DWOName that makes sense, but are you sure about DwarfOutputPath?

Reply:
Yes, ran into a similar issue for DwarfOutputPath where it was nullopt and the output path was empty as a result so the folder was missing dwo files.